### PR TITLE
GMC: adopt new common/menifests

### PR DESCRIPTION
**Pull request by KfreeZ**
## Description

This PR is a replica of #176 , since #176 has some difficulties to merge back because of the rebase issues. use this one to submit code.

this PR adopt the new manifests introduced by #173 and forget the old manifests. kept some code be comment out but not deleted in case the possible regression issue, after the old manifest deleted, these used code will be cleaned out.

this PR should've solved the issue  #138, and also be part of solution to the  #166 and #143

this PR also add support to WebRetriver service

## Issues

#138, #166, #143

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

'n/a'

## Tests

e2e regression tests passed

here is the log of searchqa result:
```
kefei@satg-opea-4node-1:~/repos/kfreez/GenAIInfra/microservices-connector$ kubectl exec client-test-87d6c7d7b-wvxf5 -n chatqa -- curl http://router-service.searchqa.svc.cluster.local:8080  -X POST \
  -d '{"text":"What is the latest news? Give me also the source link."}' \
  -H 'Content-Type: application/json'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    65    0     0  100    65      0      2  0:00:32  0:00:27  0:00:05     0data: b'\n'

data: b'\n'

data: b'An'

data: b'swer'

data: b':'

data: b' The'

data: b' latest'

data: b' news'

data: b' can'

data: b' be'

data: b' found'

data: b' at'

data: b' the'

data: b' source'

data: b' link'

data: b':'

data: b' https'

data: b'://'

data: b'www'

data: b'.'

data: b'c'

data: b'nn'

data: b'.'

data: b'com'

data: b'/,'

data: b' where'

data: b' you'

data: b' can'

data: b' view'

data: b' breaking'

data: b' news'

data: b' today'

data: b' for'

data: b' various'

data: b' categories'

data: b' such'

data: b' as'

data: b' U'

data: b'.'

data: b'S'

data: b'.,'

data: b' world'

data: b','

data: b' weather'

data: b','

data: b' entertainment'

data: b','

data: b' politics'

data: b','

data: b' and'

data: b' health'

data: b'.'

data: b'</s>'

data: [DONE]
```

### CODE REVIEW

File: .github/workflows/scripts/e2e/gmc_install.sh [hunk 1]
```diff
@@ -36,6 +36,7 @@ function copy_manifests() {
     # Copy manifest into gmc
     mkdir -p $(pwd)/config/manifests
     cp $(dirname $(pwd))/manifests/ChatQnA/*.yaml -p $(pwd)/config/manifests/
```
Comment (irisdingbj) [inline .github/workflows/scripts/e2e/gmc_install.sh:38]: remove coping from ChatQnA after all manifests finalized

**Review by irisdingbj — APPROVED** [review]

File: .github/workflows/scripts/e2e/gmc_install.sh (hunk 1 above)
Comment (KfreeZ) [reply .github/workflows/scripts/e2e/gmc_install.sh:38]: yes, that is planned

File: microservices-connector/internal/controller/gmconnector_controller.go [hunk 2]
```diff
@@ -85,119 +100,267 @@ type RouterCfg struct {
...
 	}
-	return tmpltFile
 }
 
-func reconcileResource(ctx context.Context, client client.Client, step string, ns string, svc string, svcCfg *map[string]string, retSvc *corev1.Service) error {
-	fmt.Printf("get step %s config for %s@%s: %v\n", step, svc, ns, svcCfg)
-	tmpltFile := getManifestYaml(step)
-	if tmpltFile == "" {
-		return errors.New("unexpected target")
+// func getManifestYaml(step string) string {
```
Comment (zhlsunshine) [inline microservices-connector/internal/controller/gmconnector_controller.go:112]: @KfreeZ  can these code lines be removed if they are not used?

Comment (KfreeZ) [reply microservices-connector/internal/controller/gmconnector_controller.go:112]: yes, I'm on it

File: microservices-connector/internal/controller/gmconnector_controller.go [hunk 3]
```diff
@@ -85,119 +100,267 @@ type RouterCfg struct {
...
+			return nil, fmt.Errorf("failed to reconcile resource: %v", err)
+		} else {
+			fmt.Printf("Success to reconcile %s: %s\n", obj.GetKind(), obj.GetName())
+			retObjs = append(retObjs, obj)
+		}
+	}
+	return retObjs, nil
+}
+
+func keyIsSomeEndpoint(keyname string) bool {
```
Comment (zhlsunshine) [inline microservices-connector/internal/controller/gmconnector_controller.go:262]:
```suggestion
func isDownStreamEndpointKey(keyname string) bool {
```

File: microservices-connector/internal/controller/gmconnector_controller.go [hunk 4]
```diff
@@ -85,119 +100,267 @@ type RouterCfg struct {
...
-					deployment := &appsv1.Deployment{}
-					err = runtime.DefaultUnstructuredConverter.FromUnstructured(createdObj.UnstructuredContent(), deployment)
-					if err != nil {
-						return fmt.Errorf("Failed to save deployment: %v\n", err)
-					}
-					for i := range deployment.Spec.Template.Spec.Containers {
-						deployment.Spec.Template.Spec.Containers[i].Env = append(
-							deployment.Spec.Template.Spec.Containers[i].Env,
-							newEnvVars...)
+					if keyIsSomeEndpoint(name) {
```
Comment (zhlsunshine) [inline microservices-connector/internal/controller/gmconnector_controller.go:222]:
```suggestion
					if isDownStreamEndpointKey(name) {
```

**Review by zhlsunshine — APPROVED** [review]
LGTM

**Pull request merged**